modules/javafx.controls/src/main/java/javafx/scene/control/Skin.java

```diff
@@ -60,6 +71,23 @@
      */
     public Node getNode();
 
+    /**
+     * Called by {@link Skinnable#setSkin(Skin)} after the
+     * previous skin, if any, has been uninstalled via its {@link #dispose()} method.
+     * This method allows a Skin to register listeners, add child nodes, set
+     * required properties and/or event handlers.
+     * <p>
+     * The default implementation of this method does nothing.
+     *
+     * @implNote
+     * Most implementations of Skin in the <code>javafx.controls</code> module
+     * do not need to implement {@code install} unless they must set one or more
+     * properties in the corresponding Skinnable.
+     *
+     * @since 20
+     */
+    default public void install() { }
+
```

Comment:
what about calling install more than once? There are arguments for either:

- in symmetry to dispose, it should be allowed 
- regarding this as an "out-sourced" part of the constructor - to allow/enforce proper cleanup - it shouldn't

whatever the decision, it must be specified

Reply:
hmm .. it's still using setSkin :)

What about following the lead dispose doc closely:

> Called by Skinnable when the Skin is attached/set (whatever, you are the native speakers ;) to the Skinnable. This method allows a Skin to ... (same as currently) ..
> 
> Must be called exactly once _after_ a previous Skin - if any - is disposed.
> 
> (same as current)
> 
> 

Still not quite sure the spec is clear enough to express the precondition for this method: in words it's something like "there must not be any other skin with other.getSkinnable() == this.getSkinnable()", that is all other potential skins are disposed - but that's sounds a bit unbounded ...

Aside: strictly speaking, the dispose doc is over-specified "Called by the Skinnable.." - which is making our own code violating that contract: dispose is called freely from other collaborators, f.i. during auto-sizing of columns, doing layout in VirtualFlow .. just saying :)

Reply:
re-phrased a bit, see if you like it better.